src/ray/raylet/agent_manager.cc

```diff
@@ -101,8 +105,16 @@ void AgentManager::StartAgent() {
             .WithField("pid", agent_pid_)
         << "Agent process with pid " << child.GetId() << " exit, return value "
         << exit_code;
-    RAY_UNUSED(delay_executor_([this] { StartAgent(); },
-                               RayConfig::instance().agent_restart_interval_ms()));
+    if (agent_restart_count_ < RayConfig::instance().agent_max_restart_count()) {
+      RAY_UNUSED(delay_executor_(
+          [this] {
+            agent_restart_count_++;
+            StartAgent();
+          },
+          // Retrying with exponential backoff
+          RayConfig::instance().agent_restart_interval_ms() *
+              std::pow(2, (agent_restart_count_ + 1))));
```

Comment:
Nice backoff. Can we add some logs for the restarts or `the agent_restart_count_ >= RayConfig::instance().agent_max_restart_count()`? The original logic always logs when restarting agent.

Reply:
SGTM